src/main/java/seedu/address/logic/parser/UpdateEventCommandParser.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REPEATABLE_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REPEATABLE_INTERVAL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_UPDATE_INDEX;
+
+import java.time.LocalDate;
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.UpdateEventCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.task.Interval;
+import seedu.address.model.task.repeatable.Event;
+
+/**
+ * Parses input arguments and creates a new UpdateEventCommand object
+ */
+public class UpdateEventCommandParser implements Parser<UpdateEventCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the UpdateEventCommand.
+     * and returns an UpdateEventCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    public UpdateEventCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_UPDATE_INDEX, PREFIX_DESCRIPTION,
+                PREFIX_REPEATABLE_INTERVAL, PREFIX_REPEATABLE_DATE);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_UPDATE_INDEX, PREFIX_DESCRIPTION,
```

Comment:
Parameters should be optional, refer to EditCommand.
There should be at at least one field provided.